src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,77 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String BUY = "buy";
+    private static final String SUPPLY = "supply";
+    private static final String COMMA = ",";
+    private static final String NEW_LINE = "\n";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] fileDataArray = readFile(fromFileName).split(NEW_LINE);
+        String[] arrayToBeWritten = prepareStingToWrite(fileDataArray);
+        writeData(arrayToBeWritten, clearFile(toFileName));
+    }
+
+    private String readFile(String fromFileName) {
+        String content = "";
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                content += line + "\n";
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read the file.", e);
```

[Don’t ignore exceptions](https://mate-academy.github.io/jv-program-common-mistakes/java-core/builder-file/work-with-file#dont-ignore-exceptions)